test/rhTable.js

```diff
@@ -0,0 +1,139 @@
+"use strict";
+
+var utils = require( "./utils" );
+
+describe( "Roadhouse Table", function ()
+{
+    it( "should load", function ()
+    {
+        browser.get( "/" );
+    } );
+
+    it( "should be empty without a definition", function ()
+    {
+        expect( $( "#harness" ).isPresent() ).toBe( true );
+        expect( $( "#harness .rh-table" ).isPresent() ).toBe( false );
+    } );
+
+    it( "should have no rows without a list", function ()
+    {
+        utils.set( "definition", { meta: {}, id: {}, name: {} } );
+
+        expect( $$( ".rh-table thead tr" ).count() ).toBe( 1 ); // 1 header rows
+        expect( $$( ".rh-table tbody tr" ).count() ).toBe( 0 ); // 0 body rows
+    } );
+
+    it( "should have defaults with data", function ()
+    {
+        utils.set( "list", [ { name: "one", id: 1 }, { name: "two", id: 2 } ] );
+
+        expect( $$( ".rh-table thead th" ).count() ).toBe( 3 ); // 3 columns
+        expect( $$( ".rh-table thead tr" ).count() ).toBe( 1 ); // 1 header row
+        expect( $$( ".rh-table tbody tr" ).count() ).toBe( 2 ); // 2 body rows
+        expect( $$( ".rh-table tbody tr:first-child .glyphicon-edit" ).count() ).toBe( 1 ); // edit button
+        expect( $$( ".table-controls" ).count() ).toBe( 1 ); // add button visible
+    } );
+
+    it( "should have non-view columns", function ()
+    {
+        utils.set( "definition", { meta: {}, id: { canViewList: false }, name: {} } );
+
+        expect( $$( ".rh-table thead th" ).count() ).toBe( 2 ); // 2 columns
```

Comment:
Add a check prior to setting the definition to show this action changes the number of columns.